common/interpreter/wawaka_wasm/CMakeLists.txt

```diff
@@ -93,6 +98,18 @@ if (NOT DEFINED WAMR_BUILD_LIBC_WASI)
   set (WAMR_BUILD_LIBC_WASI 0)
 endif ()
 
+# Set the memory configuration for wawaka
+IF (WASM_MEM_CONFIG STREQUAL "SMALL")
+  add_definitions (-DWASM_RUNTIME_HEAP_SIZE=2*1024*1024 -DMODULE_HEAP_SIZE=16*1024 -DMODULE_STACK_SIZE=4*1024)
+  message(STATUS "Using SMALL memory configuration")
+ELSEIF (WASM_MEM_CONFIG STREQUAL "LARGE")
+  add_definitions (-DWASM_RUNTIME_HEAP_SIZE=8*1024*1024 -DMODULE_HEAP_SIZE=2*1024*1024 -DMODULE_STACK_SIZE=1024*1024)
+  message(STATUS "Using LARGE memory configuration")
+ELSE ()
+  add_definitions (-DWASM_RUNTIME_HEAP_SIZE=4*1024*1024 -DMODULE_HEAP_SIZE=128*1024 -DMODULE_STACK_SIZE=32*1024)
+  message(STATUS "Using MEDIUM memory configuration")
+ENDIF ()
+
```

Comment:
i don't understand the rationale for these. for medium, we are giving 4M overall, with 128K to heap and 32K to stack leaving us with over 3M wasted memory. i find it very hard to believe that WAMR needs 3M to run (or if it does then we have bigger problems)

Reply:
Agreed. Given Wenyong's description, it seems like the WASM module code also takes up some of the overall heap, so I hope my follow-up questions via email will clarify this further. Even so, we would still end up with about >2M of memory unaccounted for.

Also, FWIW, they do say that the optimized interpreter uses more memory, so it might be worth looking into whether it would make sense to have interpreter-specific settings.